tools/standalone/Standalone.hs

```diff
@@ -0,0 +1,373 @@
+
+            logFunctionText l Info $ "got stats"
+            logFunctionJson logger Info stats
+            logFunctionText l Info $ "logged stats"
+            threadDelay 60000000 {- 1 minute -}
+
+node :: Logger logger => StandaloneConfiguration -> Maybe BlockStopState -> logger -> IO ()
+node conf mbs logger = do
+    rocksDbDir <- getRocksDbDir
+    when (_nodeConfigResetChainDbs conf) $ destroyRocksDb rocksDbDir
+    withRocksDb rocksDbDir $ \rocksDb -> do
+        logFunctionText logger Info $ "opened rocksdb in directory" <> sshow rocksDbDir
+        withChainwebStandalone cwConf logger rocksDb (_nodeConfigDatabaseDirectory conf) (_nodeConfigResetChainDbs conf)
+          $ \cw ->
+          stopWrapper cw $
+          mapConcurrently_ id
+            [ runChainweb' cw
+            , runCutMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
+            , runQueueMonitor (_chainwebLogger cw) (_cutResCutDb $ _chainwebCutResources cw)
+            , runRtsMonitor (_chainwebLogger cw)
+            ]
+  where
+    cwConf = _nodeConfigChainweb conf
+    nodeText = T.unpack (toText (_configNodeId cwConf))
+    v = _configChainwebVersion cwConf
+    getRocksDbDir = case _nodeConfigDatabaseDirectory conf of
+      Nothing -> getXdgDirectory XdgData
+            $ "chainweb-standalone-node/" <> sshow v <> "/" <> nodeText <> "/rocksDb"
+      Just d -> return d
+    stopWrapper cw = case mbs of
+        Nothing -> id
+        Just (Height bh) -> concurrently_ (stopAtBlockHeight bh (_cutResCutDb $ _chainwebCutResources cw))
+        Just (Weight bw) -> concurrently_ (stopAtBlockWeight bw (_cutResCutDb $ _chainwebCutResources cw))
+
+-- | Starts server and runs all network clients
+--
+runChainweb'
+    :: forall logger cas
+    . Logger logger
+    => PayloadCas cas
+    => Chainweb logger cas
+    -> IO ()
+runChainweb' cw = do
+    logg Info "start chainweb node"
+
+    -- miner
+    withAsync miner wait
+
+  where
+    logg = logFunctionText $ _chainwebLogger cw
+    miner = maybe go (\m -> runMiner (_chainwebVersion cw) m) $ _chainwebMiner cw
+        where
+          go = do
+            logg Warn "No miner configured. Starting consensus without mining."
+            forever (threadDelay 1000000)
+
+data StandaloneConfiguration = StandaloneConfiguration
+  { _nodeConfigChainweb :: !ChainwebConfiguration
+  , _nodeConfigLog :: !LogConfig
+  , _nodeConfigDatabaseDirectory :: !(Maybe FilePath)
+  , _nodeConfigResetChainDbs :: !Bool
+  , _nodeConfigStopCondition :: StopState
+  }
+  deriving (Show, Eq, Generic)
+
+makeLenses ''StandaloneConfiguration
+
+defaultStandaloneConfiguration :: ChainwebVersion -> StandaloneConfiguration
+defaultStandaloneConfiguration v = StandaloneConfiguration
+    { _nodeConfigChainweb = defaultChainwebConfiguration v
+    , _nodeConfigLog = defaultLogConfig
+        & logConfigLogger . L.loggerConfigThreshold .~ L.Info
+    , _nodeConfigDatabaseDirectory = Nothing
```

Comment:
By using the default database this node is going to create a fork for the respective chainweb version. Not sure if that's an issue. I guess it's fine.

Reply:
Given your comment, I'l change the default database to something else.